src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadataFactory.php

```diff
@@ -37,28 +34,11 @@ public function createArgumentMetadata($controller): array
         }
 
         foreach ($reflection->getParameters() as $param) {
-            $attribute = null;
             if (\PHP_VERSION_ID >= 80000) {
-                $reflectionAttributes = $param->getAttributes(ArgumentInterface::class, \ReflectionAttribute::IS_INSTANCEOF);
-
-                if (\count($reflectionAttributes) > 1) {
-                    $representative = $controller;
-
-                    if (\is_array($representative)) {
-                        $representative = sprintf('%s::%s()', \get_class($representative[0]), $representative[1]);
-                    } elseif (\is_object($representative)) {
-                        $representative = \get_class($representative);
-                    }
-
-                    throw new InvalidMetadataException(sprintf('Controller "%s" has more than one attribute for "$%s" argument.', $representative, $param->getName()));
-                }
-
-                if (isset($reflectionAttributes[0])) {
-                    $attribute = $reflectionAttributes[0]->newInstance();
-                }
+                $attributes = $param->getAttributes();
```

Comment:
I've been wondering about this and honeslty I'm not sure we should do it:
right now, the argument-resolver subsystem s decoupled from reflection.
This means it's possible to instantiate it without using reflection at all.
This creates a coupling that cannot be worked around, as creating a standalone ReflectionAttribute instance is not possible afaik (Am I wrong?)

Reply:
Right, reverted back to returning attribute instances, with a `class_exists()` check ignoring undefined ones. Makes sense?